src/ipfsd-daemon.js

```diff
@@ -246,13 +250,21 @@ class Daemon {
     let output = ''
     this.subprocess = run(this, args, { env: this.env }, {
       error: (err) => {
+        let line = String(err)
```

Comment:
The name `line` seems kind of unrelated. What do you think of `errString`, or something more related to the string content?

Reply:
`err` is a string with a newline character at the end, hence `line`.

`errString` to me would be better as `errorString` but then I've never been a fan of Hungarian notation so I don't think we should encode the type in variable names and certainly not in a language without types, so just `error`.

...But then by convention we'd expect `error` to be an `Error` object, so it confuses the purpose of this callback.  I'd prefer the options to be:

```javascript
{
  onStdErr: (line) => {},
  onStdOut: (line) => {}
}
```

instead of:

```javascript
{
  error: (err) => {},
  data: (data) => {}
}
```

Reply:
I see your point. I would suggest changing it to `errorLine` or keep as you have it.